### ISSUE TITLE
In custom remotes, error-callback will insert `ui/fetch-state` in-place even if `:marker` is set to a keyword

### ISSUE THREAD

**Issue by levitanong** [fulcrologic/fulcro#126]
```clj
;; initial app-state
{...
 :app/things []
 }

;; after a load error
{...
 :app/things {:ui/fetch-state {...}}
 }

;; this messes things up because app/things is a join against an identified component.
[{:app/things (prim/get-query Thing)}]

;; so the parent component receives a weird value instead of []
```

**Comment by awkay** [comment]
Attempt at fix in 2.1.4-SNAPSHOT. Please test and let me know.

**Comment by levitanong** [comment]
That seems to have fixed the issue, but I'm getting these errors now:

```
[  1.484s] [fulcro.client] Mutation fulcro.inspect.ui.transactions/add-tx failed with exception Error: Assert failed: Path [:fulcro.inspect.ui.transactions/tx-list-id [:fulcro.inspect.core/app-id sample-app.ui.root/Root] :fulcro.inspect.ui.transactions/tx-list] for append must target an app-state vector.
(vector? (get-in state data-path))
```

**Comment by awkay** [comment]
That's inspect, not fulcro

**Comment by awkay** [comment]
Fixed in 203e834c76d0f1d5056174da136d99fdc0737bd7
